DESCRIPTION

RefDirectory: Throw exception if committing packed-refs fails

This should never happen, because this segment of the code is protected
by the packed-refs lock file on disk. But if it does happen, it's better
to let the caller know than to silently drop the whole commit operation
on the floor.

The existing concurrentOnlyOneWritesPackedRefs test is nondeterministic
as written. Fixing it is beyond the scope of this change, but I did
verify that it passes consistently with:

  bazel test --runs_per_test=200 //org.eclipse.jgit.test:org_eclipse_jgit_internal_storage_file_GcPackRefsTest

Change-Id: I5ff4dc39ee05bda88d47909acb70118f3d0c8f74


COMMENTS

author: Santos Moore
date: 2017-07-05 19:52:16.000000000

Uploaded patch set 1.

-------------------------------------
author: Brenden Conley
date: 2017-07-05 20:27:31.000000000

Patch Set 1:

Build Started https://hudson.eclipse.org/jgit/job/jgit.gerrit/10410/

-------------------------------------
author: Brenden Conley
date: 2017-07-05 20:58:22.000000000

Patch Set 1: Verified+1

Build Successful 

https://hudson.eclipse.org/jgit/job/jgit.gerrit/10410/ : SUCCESS

-------------------------------------
